script.js

```diff
@@ -255,43 +251,68 @@ function popUp() {
 }
 
 function getButtons() {
-  projectspopup = document.getElementsByClassName("open-modal");
+  projectspopup = document.getElementsByClassName('open-modal');
   projectspopup = Array.from(projectspopup);
-  projectspopup.forEach((e) => e.addEventListener("click", popUp));
+  projectspopup.forEach((e) => e.addEventListener('click', popUp));
 }
 
-const closeMobilePopup = document.querySelector(".desktop-popcontainer");
-const closeDesktopPopup = document.querySelector(".mobile-popcontainer");
-const closeMobPopup = document.querySelectorAll(".exit-button");
+const closeMobilePopup = document.querySelector('.desktop-popcontainer');
+const closeDesktopPopup = document.querySelector('.mobile-popcontainer');
 
 function closeButton() {
-  closeMobilePopup.style.display = "none";
-  closeDesktopPopup.style.display = "none";
-  document.body.style.overflow = "hidden";
+  closeMobilePopup.style.display = 'none';
+  closeDesktopPopup.style.display = 'none';
+  document.body.style.overflow = 'hidden';
 }
 
-closeMobPopup.forEach((a) => a.addEventListener("click", closeButton));
-window.addEventListener("click", (e) => {
-  if (e.target === closeDesktopPopup || e.target === closeMobilePopup) {
+window.addEventListener('click', (e) => {
+  if (
+    e.target === closeDesktopPopup
+    || e.target === closeMobilePopup
+    || e.target.parentNode.classList.contains('exit-button')
+  ) {
     closeButton();
   }
 });
 
-const form = document.querySelector(".contact-form");
-const span = document.querySelector(".error-message");
+const form = document.querySelector('.contact-form');
+const span = document.querySelector('.error-message');
 
 function hasUpperCase(input) {
   return input.toLowerCase() !== input;
 }
-form.addEventListener("submit", (event) => {
+form.addEventListener('submit', (event) => {
   if (hasUpperCase(form.elements.email.value)) {
     event.preventDefault();
-    span.style.display = "block";
-    span.innerHTML = "The email field only accepts lowercase letters";
+    span.style.display = 'block';
+    span.innerHTML = 'The email field only accepts lowercase letters';
   }
 });
 
 window.onload = () => {
   myWorkSection();
   getButtons();
 };
+
+let formData = {
+  name: '',
+  email: '',
+  message: '',
+};
+const fetchLocalData = () => {
+  if (localStorage.getItem('formData') !== null) {
+    formData = JSON.parse(localStorage.getItem('formData'));
+  }
+  form.elements.name.value = formData.name;
+  form.elements.email.value = formData.email;
+  form.elements.message.value = formData.message;
+};
+
+fetchLocalData();
+document.querySelector('#get-in-touch').addEventListener('click', () => {
+  formData.name = form.elements.name.value;
+  formData.email = form.elements.email.value;
+  formData.message = form.elements.message.value;
+
+  localStorage.setItem('formData', JSON.stringify(formData));
+});
```

Comment:
- [ ] Good job implementing the preserve the data in the storage function. However, when the data is not preserved when the data is not submitted. kindly fix this.

Reply:
Hello @phelian23, can you please give us an explanation of this requirement? We are finding it difficult to understand the statement.
Thank you

Reply:
- [ ] Good job @Kwentsir and @bizip by implementing the requests from the previous reviewer. Now your data is well preserved in the local storage and updated if there are any changes.